docs: enhance faq (#387)
EndstoneMC/endstone: docs/getting-started/faq.md

    Endstone is built around the official [Bedrock Dedicated Server (BDS)](https://www.minecraft.net/download/server/bedrock), which supports Windows and Ubuntu Linux.

    Accordingly, Endstone only supports Windows and Ubuntu Linux. Other platforms may or may not function.

    BDS natively supports the `x86-64` CPU architecture. Systems with `arm` CPU architectures are not officially supported.

!!! tip "ARM Architecture (Apple Silicon, etc.)"

>     While BDS is natively built for `x86-64`, it can run exceptionally well on `arm` systems using Docker emulation. In our testing on an **Apple M1 running macOS Tahoe (26.3.1)**, performance remains stable with little-to-no latency impact!

Reviewer: Would this perhaps be better as a footnote? Using an admonition makes it appear quite prominently:

<img width="793" height="461" alt="Admonition" src="https://github.com/user-attachments/assets/73b5626c-ee2b-43a9-a206-7267b65e4f6c" />

---

Using a footnote looks like this:

<img width="793" height="428" alt="Image" src="https://github.com/user-attachments/assets/a4a07c96-b275-4f50-adae-0ea947041aea" />

```suggestion
    BDS natively supports the `x86-64` CPU architecture. Systems with `arm` CPU architectures are not officially supported.[^1]

[^1]: While BDS is natively built for `x86-64`, it can run exceptionally well on `arm` systems using Docker emulation. In our testing on an **Apple M1 running macOS Tahoe (26.3.1)**, performance remains stable with little-to-no latency impact!
```

    BDS natively supports the `x86-64` CPU architecture. Systems with `arm` CPU architectures are not officially supported.[^1]

> [^1]: While BDS is natively built for `x86-64`, it can run exceptionally well on `arm` systems using [Docker emulation]. In our testing on an **Apple M1 running macOS Tahoe (26.3.1)**, performance remains stable with little-to-no latency impact! 

Reviewer: > In our testing on an **Apple M1 running macOS Tahoe (26.3.1)**

This implies that the testing was done by the Endstone project maintainers. Is that the intention here?

Author: I guess it's easy to see it that way, I didn't really think about it that much because I didn't really think it really mattered who tested it as long as it worked.

Do you have any changes in mind? I am unsure about how I should go about addressing who did the testing.

Reviewer: Removing the second sentence will still leave the implication that it works well on arm architecture. It's not strictly necessary to have a section addressing the testing.

It's implied that documentation is telling the truth, so we don't need to always give evidence.

```suggestion
[^1]: While BDS is natively built for `x86-64`, it can also run well on `arm` systems via [Docker emulation]. Using emulation can also allow Endstone to run on macOS.
```

> [Docker emulation]: installation#with-docker

Reviewer: This'll make it open the tab as well.
```suggestion
[Docker emulation]: installation.md/#with-docker-latest-arm-with-emulation
```

It gives this `INFO` log which I can't do much about. The link to the tab still works correctly though.
```
INFO    -  Doc file 'getting-started/faq.md' contains a link 'installation.md#with-docker-latest-arm-with-emulation',
           but the doc 'getting-started/installation.md' does not contain an anchor
           '#with-docker-latest-arm-with-emulation'.
```
